### PR TITLE
Add SIRIUS annotations and mz_rt fallback to MZMine converter (three-tier ProteinName)

### PR DESCRIPTION
**Pull request by swaraj-neu**
- Add SIRIUS annotation support and mz_rt fallback to MZMine converter
- Retain features lacking MZMine compound names, assign ProteinName via MZMine/SIRIUS/mz_rt tiers, restore required mz/RT metadata, and add validation, tests, docs, and tier-level logging.

# Motivation and Context

Please include relevant motivation and context of the problem along with a short summary of the solution.

## Changes

Please provide a detailed bullet point list of your changes.

- 

## Testing

Please describe any unit tests you added or modified to verify your changes.

## Checklist Before Requesting a Review
- [ ] I have read the MSstats [contributing guidelines](https://github.com/Vitek-Lab/MSstatsConvert/blob/master/.github/CONTRIBUTING.md)
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

## Motivation and Context

The PR adds support for SIRIUS structural annotations and implements a fallback mechanism for the MZMine converter in MSstatsConvert. Previously, the converter would drop features lacking MZMine spectral-library annotations (MSI Level 2). This enhancement introduces a three-tier `ProteinName` assignment strategy: (1) MZMine compound names (highest priority), (2) SIRIUS structure identifications when available (MSI Level 3, in-silico structure prediction), and (3) a synthesized m/z and retention time identifier as a final fallback. This approach retains all features while maintaining annotation precedence and improving metabolomics data coverage.

### Implementation Changes

- **`R/clean_MZMine.R` (`.cleanRawMZMine` function)**
  - Added optional `sirius_annotations` parameter (default `NULL`)
  - Changed `ProteinName` assignment from a single inner join (which discarded unmatched features) to a three-tier left-join strategy:
    - Tier 1: Left join MZMine `compound_name` by `rowID` → `id` (features with MZMine annotations receive highest-scoring compound name)
    - Tier 2: If `sirius_annotations` provided, fill remaining `NA` values using `mappingFeatureId` → `rowID` join with SIRIUS `name` (non-empty names only)
    - Tier 3: For any remaining `NA` features, synthesize `ProteinName` from m/z and retention time (format: `mz_rt`, e.g., `489.334_7.89` constructed as `round(mz, 4)_round(rt, 2)`)
  - Added validation for required metadata columns: `rowID`, `rowmz`, `rowretentiontime`
  - Added per-tier logging to report counts of features assigned at each tier (MZMine, SIRIUS, m/z-RT fallback) with messages like "** MZMine ProteinName assignment: MZMine compound: N feature(s); SIRIUS name: N feature(s); m/z-RT fallback: N feature(s)."
  - Updated roxygen documentation to describe new parameter and three-tier assignment strategy
  - Lines changed: +54/-16

- **`R/converters_MZMinetoMSstatsFormat.R` (public wrapper function)**
  - Added optional `sirius_annotations` parameter (default `NULL`) to function signature
  - Added validation to check that `sirius_annotations` (when non-NULL) contains required columns `mappingFeatureId` and `name`; stops with error message if missing required columns
  - Added early validation to ensure `mzmine_annotations` is provided (not NULL or missing); raises error with instruction message
  - Modified preprocessing call to pass `sirius_annotations` into `MSstatsConvert::MSstatsClean` (was previously called without it)
  - Changed `remove_single_feature_proteins` behavior: now always set to `FALSE` rather than driven by `removeProtein_with1Feature` flag
  - Extended roxygen examples to demonstrate reading SIRIUS `structure_identifications.tsv` and calling the converter with `sirius_annotations`
  - Updated `\details` section to describe the three-tier strategy, MSI level assignments, and explicit clarification that features are retained via m/z-RT fallback rather than filtered out
  - Included discussion of trade-offs: retaining all features improves normalization stability but increases hypothesis burden; recommends confirmatory users filter to MZMine-only, discovery users benefit from additional sources
  - Lines changed: +60/-15

- **Documentation Updates**
  - **`man/MSstatsClean.Rd`**: Updated S4 method signature to accept `sirius_annotations = NULL`; clarified that SIRIUS `name` populates `ProteinName` for features without MZMine compound names; notes schema validated against SIRIUS 6 output (lines: +13/-6)
  - **`man/MZMinetoMSstatsFormat.Rd`**: Documented new `sirius_annotations` parameter; rewrote `\details` section to describe three-tier assignment with MSI level context (Level 2 for MZMine, Level 3 for SIRIUS); updated examples with SIRIUS demonstration; clarified features are retained rather than filtered; explained m/z-RT fallback format (lines: +64/-15)
  - **`man/dot-cleanRawMZMine.Rd`**: Documented `sirius_annotations` parameter; specified only `mappingFeatureId` and `name` columns are read; noted score/confidence columns are ignored; clarified schema validation against SIRIUS 6 with guidance for other versions; removed prior statement that features without matching annotations are dropped (lines: +13/-6)
  - **`vignettes/msstats_data_format.Rmd`**: Updated vignette section to explain three-tier strategy with precedence, SIRIUS schema expectations (`mappingFeatureId` matches to MZMine `rowID`), and replaced prior behavior (drop features; no fallback) with new behavior (retain via m/z-RT fallback); extended worked examples to show both baseline conversion and SIRIUS-enriched conversion followed by inspection of unique `PeptideSequence`/`ProteinName` pairs (lines: +52/-12)

### Unit Tests Added/Modified

- **`inst/tinytest/test_converters_MZMinetoMSstatsFormat.R`** (lines: +50/-22)
  - **Baseline MZMine annotation test**:
    - Verified all 6 features retained (24 rows = 6 features × 4 runs) instead of dropping unannotated features
    - Confirmed features 4 and 5 receive m/z-RT fallback `ProteinName` values (`489.334_7.89`, `555.447_9.1`) when absent from `mzmine_annotations`
    - Verified correct handling of feature 2 (highest-scoring MZMine annotation wins among duplicates; feature 2 with two annotation rows correctly assigned "GlucoseHigh")
    - Tested standard column presence and structure (11 columns, correct column names)
    - Tested data type handling (IsotopeLabelType set to "Light" for metabolomics, charge/fragment columns NA, Fraction set to 1)
    - Tested zero-intensity input cells converted to NA in output
    - Verified annotation merging with run metadata (Condition, BioReplicate correctly joined)
    - Tested intensity values traced back correctly to input data
  
  - **SIRIUS annotation test**:
    - Tested precedence: MZMine `compound_name` (Caffeine) beats SIRIUS for feature 1 (confirmed both sources available but MZMine wins)
    - Verified SIRIUS name fill: feature 4 (no MZMine match) assigned SIRIUS name `"Caffeic acid"`
    - Tested m/z-RT fallback when SIRIUS provides empty/invalid name: feature 5 with only empty-name SIRIUS row correctly falls back to `"555.447_9.1"`
    - Verified irrelevant SIRIUS rows (mappingFeatureId=99) do not introduce spurious new features or protein names ("Ghost")
  
  - **Error handling tests**:
    - Confirmed `mzmine_annotations = NULL` raises "mzmine_annotations is required" error with guidance message
    - Confirmed omitting `mzmine_annotations` argument (missing) raises same error
    - Confirmed malformed `sirius_annotations` (missing required `name` column) triggers `stop()` with "missing required column" error message

## Coding Guidelines Compliance

No explicit violations of coding guidelines were identified in the provided changes. The implementation demonstrates:
- Consistent roxygen documentation practices with proper parameter documentation, details sections, and examples
- Proper use of data.table idioms (left joins, `setorder`, `unique`, `set` for column deletion) consistent with codebase patterns
- Appropriate input validation with informative error messages for missing or malformed inputs
- Informative logging at multiple tiers to document feature assignment decisions via `getOption("MSstatsLog")` and `getOption("MSstatsMsg")`
- Comprehensive updates to all related documentation (roxygen, man pages, vignettes) in coordination with code changes
- Well-structured test coverage for normal operation, edge cases, and error conditions

### CODE REVIEW

File: R/converters_MZMinetoMSstatsFormat.R [hunk 1]
```diff
@@ -14,16 +14,45 @@
...
+#'   MS/MS spectral matching).
+#' @param sirius_annotations Optional `data.frame` of SIRIUS
+#'   `structure_identifications.tsv` output, or `NULL`. Only the
+#'   `mappingFeatureId` and `name` columns are read; score columns
+#'   (`ConfidenceScoreExact`, `ConfidenceScoreApproximate`,
+#'   `SiriusScore`) are ignored in this release. When supplied, SIRIUS
+#'   `name` fills `ProteinName` for features that received no MZMine
+#'   compound (tier 2, MSI Level 3 in-silico structure prediction).
+#'   The schema is validated against SIRIUS 6 output; users on other
+#'   versions can rename columns to match. Pass `NULL` to disable.
```
Comment (tonywu1999) [inline R/converters_MZMinetoMSstatsFormat.R]: Could this use .inheritParams from the .cleanRawMZMine so that we don't duplicate documentation?

File: R/converters_MZMinetoMSstatsFormat.R [hunk 2]
```diff
@@ -14,16 +14,45 @@
...
+#'
+#' 2. **SIRIUS name (optional)** -- when `sirius_annotations` is
+#'    non-NULL, the `name` from SIRIUS `structure_identifications.tsv`
+#'    fills any `ProteinName` still NA after tier 1. Equivalent to MSI
+#'    Level 3: in-silico structure prediction. MZMine annotations take
+#'    precedence: SIRIUS only fills features that MZMine missed.
+#'
+#' 3. **mz_rt fallback (always)** -- features with no annotation from
+#'    either source are retained, not dropped, and assigned
+#'    `paste0(round(mz, 4), "_", round(rt, 2))` as their `ProteinName`.
```
Comment (tonywu1999) [inline R/converters_MZMinetoMSstatsFormat.R]: Probably should have details more user friendly to a biologist than `paste0(round(mz, 4), "_", round(rt, 2))`

File: R/converters_MZMinetoMSstatsFormat.R [hunk 3]
```diff
@@ -14,16 +14,45 @@
...
+#'
+#' The tier-3 retain-all policy is a deliberate trade-off. A fuller
+#' feature set gives more stable medians and a more reliable empirical
+#' distribution for global normalization. SIRIUS extends discovery
+#' coverage to features that level-2 spectral matching misses. The
+#' cost is an increase in the number of hypotheses tested downstream
+#' (in `MSstats::groupComparison`), which weakens multiple-testing
+#' correction. Users running confirmatory analyses should restrict to
+#' tier-1 features post-conversion; users running discovery analyses
+#' benefit from the additional tiers despite the FDR burden.
```
Comment (tonywu1999) [inline R/converters_MZMinetoMSstatsFormat.R]: There seems to be confusing language

tier-1, tier-2, level-2, level-3, etc.  I'd stick with just mentioning levels, and don't mention anything about tiers.  This would apply in your vignette as well.

File: vignettes/msstats_data_format.Rmd [hunk 4]
```diff
@@ -353,17 +353,50 @@ per feature, one `<sample> Peak area` column per sample) together with a standar
...
+deliberate trade-off:
+
+- **Pro**: a fuller feature set gives more stable medians and a more
+  reliable empirical distribution for global normalization.
+- **Pro**: SIRIUS extends discovery coverage to features that level-2
+  spectral matching misses.
+- **Con**: retaining all features increases the number of hypotheses
+  tested downstream, which weakens multiple-testing correction. For
+  confirmatory analyses, restrict to tier-1 features after conversion.
+- **Note**: passing `removeProtein_with1Feature = TRUE` drops nearly
```
Comment (tonywu1999) [inline vignettes/msstats_data_format.Rmd]: Let's get rid of this parameter `removeProtein_with1Feature` from the function for now.  Wherever it's needed later on, hard-code to FALSE.

File: R/clean_MZMine.R [hunk 5]
```diff
@@ -57,20 +65,50 @@
...
-    retained_msg <- paste0("** MZMine: retained ", length(retained_ids),
-                           " feature(s) after annotation join: ",
-                           paste(retained_ids, collapse = ", "))
-    getOption("MSstatsLog")("INFO", retained_msg)
-    getOption("MSstatsMsg")("INFO", retained_msg)
+    # Tier 2: SIRIUS name fills features still NA after tier 1.
+    n_tier2 <- 0L
+    if (!is.null(sirius_annotations)) {
+        sirius_dt <- data.table::as.data.table(sirius_annotations)
+        sirius_dt <- sirius_dt[, c("mappingFeatureId", "name"), with = FALSE]
```
Comment (tonywu1999) [inline R/clean_MZMine.R]: Apparently there's a more RAM friendly way to take subsets of columns in a data.table.  Right now, this code duplicates the table (I don't think this table will ever get that large where that's a problem, but putting this here so that in the future, we follow this coding practice)

```
drop_cols = setdiff(colnames(sirius_dt), c("mappingFeatureId", "name"))
for (col in drop_cols) data.table::set(sirius_dt, j = col, value = NULL)
```

File: R/clean_MZMine.R [hunk 6]
```diff
@@ -57,20 +65,50 @@
...
+        data.table::setorder(sirius_dt, mappingFeatureId)
+        sirius_dt <- unique(sirius_dt, by = "mappingFeatureId")
+        mz_input[, ProteinName := ifelse(
+            is.na(ProteinName),
+            sirius_dt[
+                .(as.character(get(id_col))),
+                on = "mappingFeatureId",
+                name
+            ],
+            ProteinName)]
```
Comment (tonywu1999) [inline R/clean_MZMine.R]: This approach will copy the whole data.table in RAM.  Here's the more memory-efficient approach that will do copy in place:

```
mz_input[is.na(ProteinName), ProteinName := 
            sirius_dt[
                .(as.character(get(id_col))),
                on = "mappingFeatureId",
                name
            ]]
```

Comment (tonywu1999) [reply R/clean_MZMine.R]: Haven't tested that the code above works, but here's an example where my intuition comes from:

```
# copying
input[, nonmissing_orig := ifelse(is.na(newABUNDANCE), TRUE, nonmissing_orig)]

# modifying in place
input[is.na(newABUNDANCE), nonmissing_orig := TRUE]
```

File: R/clean_MZMine.R [hunk 7]
```diff
@@ -57,20 +65,50 @@
...
+    # Tier 2: SIRIUS name fills features still NA after tier 1.
+    n_tier2 <- 0L
+    if (!is.null(sirius_annotations)) {
+        sirius_dt <- data.table::as.data.table(sirius_annotations)
+        sirius_dt <- sirius_dt[, c("mappingFeatureId", "name"), with = FALSE]
+        sirius_dt[, name := ifelse(is.na(name) | name == "",
+                                   NA_character_, as.character(name))]
+        sirius_dt[, mappingFeatureId := as.character(mappingFeatureId)]
+        data.table::setorder(sirius_dt, mappingFeatureId)
+        sirius_dt <- unique(sirius_dt, by = "mappingFeatureId")
```
Comment (tonywu1999) [inline R/clean_MZMine.R:91]: Is this line of code needed if you're doing a mapping-like operation in the next line?  Is it to account for cases when there's multiple names for a single mappingFeatureId?

File: R/converters_MZMinetoMSstatsFormat.R [hunk 8]
```diff
@@ -13,17 +14,42 @@
 #'   so the corresponding `Run` value must be `sampleAmzML`.
 #' @param mzmine_annotations `data.frame` of MZMine spectral-library
 #'   annotations with columns `id`, `compound_name`, `score`. Required:
-#'   the highest-scoring `compound_name` per feature is used as
-#'   `ProteinName`, and features in the quant table with no matching
-#'   annotation row are dropped from the output.
+#'   the highest-scoring `compound_name` per feature (MSI Level 2
+#'   putative identification via MS/MS spectral matching) is used as
+#'   `ProteinName`.
```
Comment (tonywu1999) [inline R/converters_MZMinetoMSstatsFormat.R]: I think you can remove mzmine_annotations in the docs here if you're using inheritParams now from .cleanRawMZMine

**Review by tonywu1999 — APPROVED** [review]
